libraries/botbuilder-dialogs/src/dialogHelper.ts

```diff
@@ -113,8 +118,21 @@ function sendEoCToParent(context: TurnContext): boolean {
     return false;
 }
 
+// Helper to send a trace activity with a memory snapshot of the active dialog DC.
+export async function sendStateSnapshotTrace(dc: DialogContext, traceLabel: string): Promise<void> {
```

Comment:
I don't think we need to export this from dialogHelper since it's only used in DialogManager in C#, can you move it to mirror the C# implementation and can you fix the docstring here?

Reply:
@stevengum do you mean we need to make it a private method of DialogManager? We have several functions in dialog helper designed to be imported and used in DialogManager, they won't be exported to the outside of package, do we need to merge all of them with DialogManager?

Reply:
This function is "half-public" in the sense that if someone imported this specific file, they would have access to the function, even though the function is not exported from the index.ts. Since the function seems to be only used in DialogManager, it should live there for now as a private method.

I might be missing some references to this function, but in the PR Ctrl+F only returns 3 references besides the declaration; the import and its usage in DialogManager.

Reply:
Got it, moved `sendStateSnapshotTrace` into dialog manager.